modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -384,6 +402,45 @@ def test_loc(data):
         modin_df.loc["NO_EXIST"]
 
 
+@pytest.mark.parametrize("equal_axes", [False, True])
+@pytest.mark.parametrize("axis", [0, 1])
+def test_loc_4456_1(axis, equal_axes):
+    if equal_axes and axis == 1:
+        # case equal to `equal_axes and axis == 0`
+        pass
+
+    data = test_data["float_nan_data"]
+    modin_df, pandas_df = pd.DataFrame(data), pandas.DataFrame(data)
+
+    # prepare `loc` operands
+    key = pandas_df.columns[:2]
+    index, columns = pandas_df.index, key
+    value = np.random.randint(0, 100, size=(pandas_df.shape[0], len(columns)))
+    if not equal_axes:
+        if axis == 0:
+            index = list(reversed(index))
+        else:
+            columns = list(reversed(columns))
+
+    loc_key = (slice(None), key)
+
+    # pandas DataFrame
+    df_value = pandas.DataFrame(value, index=index, columns=columns)
+    eval_loc(modin_df, pandas_df, df_value, loc_key)
+
+    # modin DataFrame
+    mdf_value = pd.DataFrame(df_value)
+    eval_loc(modin_df, pandas_df, (mdf_value, df_value), loc_key)
```

Comment:
It just seems a bit strange to me that there's no cases for `len(key) != len(value)` as it was originally a reproducer one :) For now, it's obvious that the made changes should fix those cases as well, but it could be no longer true as our code-base is constantly changing at high-levels. So I would propose to add these cases as well.

I've rewritten this test in a separate branch so it include all of the cases. You can inspect and cherry-pick it here if you would like: https://github.com/dchigarev/modin/commit/67352aad5ceefeecb188c09ed92d11066cbf8387